contracts/test/StakeLocker.t.sol

```diff
@@ -2,663 +2,254 @@
+        staker.stakeLocker_stake(address(locker), stakeAmount);
+        staker.stakeLocker_intendToUnstake(address(locker));
 
-        uint256 cReq = loan1.collateralRequiredForDrawdown(10_000_000 * USD);  // WETH required for 100_000_000 USDC drawdown on loan
-        mint("WETH", address(bob), cReq);                                      // Mint WETH to borrower
-        bob.approve(WETH, address(loan1), MAX_UINT);                           // Borrower approves WETH
-        bob.drawdown(address(loan1), 10_000_000 * USD);                        // Borrower draws down 10m USDC
+        uint256 start = block.timestamp;
 
-        mint("USDC", address(bob), 10_000_000 * USD);  // Mint USDC to Borrower for repayment plus interest
-        bob.approve(USDC, address(loan1), MAX_UINT);   // Borrower approves USDC
-        bob.makeFullPayment(address(loan1));           // Borrower makes full payment, which includes interest
+        // Can't unstake immediately (lockup not expired)
+        assertTrue(!staker.try_stakeLocker_unstake(address(locker), unstakeAmount));
 
-        pat.claim(address(pool1), address(loan1), address(dlFactory1));  // PD claims interest, distributing funds to stakeLocker
-    }
+        hevm.warp(start + globals.stakerCooldownPeriod() - 1);
 
-    function test_unstake(uint256 stakeAmount) public {
-        uint256 bptMin = WAD / 10_000_000;
-        stakeAmount = constrictToRange(stakeAmount, bptMin, bPool.balanceOf(address(sam)), true);  // 25 WAD max, 1/10m WAD min, or zero (min is roughly equal to 10 cents) (non-zero)
+        // Can't unstake in cooldown (lockup not expired)
+        assertTrue(!staker.try_stakeLocker_unstake(address(locker), unstakeAmount));
 
-        uint256 stakeDate = block.timestamp;
+        hevm.warp(start + globals.stakerCooldownPeriod() + (globals.stakerUnstakeWindow() / 2));
 
-        makePublicAndStake(stakeAmount);
+        // Can't unstake in unstake window yet (lockup not expired)
+        assertTrue(!staker.try_stakeLocker_unstake(address(locker), unstakeAmount));
 
-        assertEq(IERC20(USDC).balanceOf(address(sam)),                          0);
-        assertEq(bPool.balanceOf(address(sam)),          (25 * WAD) - stakeAmount);
-        assertEq(bPool.balanceOf(address(stakeLocker1)), (50 * WAD) + stakeAmount);  // PD + Staker stake
-        assertEq(stakeLocker1.totalSupply(),             (50 * WAD) + stakeAmount);
-        assertEq(stakeLocker1.balanceOf(address(sam)),                stakeAmount);
-        assertEq(stakeLocker1.stakeDate(address(sam)),                  stakeDate);
+        staker.stakeLocker_cancelUnstake(address(locker));
 
-        setUpLoanAndRepay();
-        assertTrue(!sue.try_intendToUnstake(address(stakeLocker1)));  // Unstake will not work as sue doesn't possess any balance.
-        assertTrue( sam.try_intendToUnstake(address(stakeLocker1)));
+        hevm.warp(start + 180 days + 1);
```

Comment:
Should be `hevm.warp(start + globals.lockupPeriod());` to make things clearer, also don't need +1 since condition is `<=`

Reply:
But the `StakerLocker` constructor is what defines `lockupPeriod   = 180 days`, not globals.

Reply:
Yeah my bad, I realize after that its `locker.lockupPeriod()`